### PR TITLE
Bug - failing with exception while checking the estimated row count

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultCustomAggregateService.java [hunk 1]
```diff
@@ -37,30 +37,31 @@
...
         authorizationService.checkAuthorization(user, query);
-        checkIsValid(query);
-        return repository.findByQuery(user, ensureDimension(query, Overall));
+        return repository.findByQuery(user, checkIsValid(query));
     }
 
     @Override
     public int estimateReportRowCount(final User user, final ReportQuery query) {
         authorizationService.checkAuthorization(user, query);
-        checkIsValid(query);
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultCustomAggregateService.java]: Front end does not sent Overall, and the check for valid was failing.

**Comment by agorina** [comment]
Merging, will address comments if needed.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultCustomAggregateService.java (hunk 1 above)
Comment (malaffoon) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultCustomAggregateService.java]: Just for my benefit, what is the actual change? It looks like the same calls are being made and the code was just pushed around a bit?

Comment (agorina) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultCustomAggregateService.java]: `ensureDimension` adds Overall dimension to the query if it is missing. Before this change it was done after `checkIsValid`, causing the check to fail.

Comment (malaffoon) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultCustomAggregateService.java]: Ah, thanks.

**Pull request merged**